### PR TITLE
Report invalid units generated during test session (bugfix)

### PR DESCRIPTION
**Pull request by Hook25**
## Description

Currently if a template generates a unit that is invalid, a message is logged (error for some, debug for some others) and the unit is skipped. This message is visible using Checkbox local but it burried in the Agent log in remote. This is of course sub-optimal as we have no recollection of anything happening after the fact but this can lead to tests being... not executed, without an explanation or even a way for the user to notice. 

This PR introduces a new unit, the `InvalidJob` unit, that allows us to report the invalidly generated units as failures, so that they are present all reports and on C3. This change is made along the new `feature` section, that allows us to toggle features like this on or off (to allow people to transition into big changes instead of having to update forcibly). The changes are both in this PR because the feature change is about 4 lines, and it makes sense to create and use it (else, I don't see how I could test it)

## Resolved issues

Fixes: CHECKBOX-1677
Fixes: CHECKBOX-1676

## Documentation

Pending implementation approval, I think this should be mentioned in both the extended tutorial and the normal tutorial

## Tests

This implements a new metabox scenario and unit tests

### CODE REVIEW

File: checkbox-ng/plainbox/impl/unit/test_unit.py [hunk 1]
```diff
@@ -203,8 +203,14 @@ def test_get_record_value(self):
         self.assertEqual(unit1.get_record_value("key"), "value")
         self.assertEqual(unit2.get_record_value("key"), "value")
         self.assertEqual(unit3.get_record_value("key"), "value")
-        with self.assertRaises(MissingParam):
+        with self.assertRaises(MissingParam) as mp:
+            # TODO: this is structured badly, if a unit has a template-engine
+            #       it is a tempalte, but the definition is in Unit, not
```
Comment (pieqq) [inline checkbox-ng/plainbox/impl/unit/test_unit.py:208]:
```suggestion
            #       it is a template, but the definition is in Unit, not
```

**Review by pieqq — CHANGES_REQUESTED** [review]
Great stuff!

A few typos and other small comments, but otherwise good to land I think!

File: checkbox-ng/plainbox/impl/config.py [hunk 2]
```diff
@@ -389,6 +389,17 @@ class DynamicSection(dict):
             ),
         },
     ),
+    (
+        "features",
+        {
+            "strict_template_expansion": VarSpec(
+                bool,
+                True,
+                "Units generated by templates can crash Checkbox when invalid or "
```
[the remark this answers is not in the document]
Comment (Hook25) [reply checkbox-ng/plainbox/impl/config.py]: Fixed

File: checkbox-ng/plainbox/impl/config.py [hunk 3]
```diff
@@ -389,6 +389,17 @@ class DynamicSection(dict):
...
+                "be reported as invalid",
```
[the remark this answers is not in the document]
Comment (Hook25) [reply checkbox-ng/plainbox/impl/config.py]: Improved the description, see if makes sense

**Review by pieqq — APPROVED** [review]
Much clearer, thanks!

**Pull request merged**